feat(UI): Always put post body behind toggle on post listings

jsit opened the pull request:
Completes #1595

Thoughts on this? The body showing up in post listings was occupying a lot of vertical space, and from the sound of it, people don't like this.

With this PR:

1. On a Listing Page (home/community), there is an expand/collapse icon next to the title that will show both (a) the post body, and (b) the link preview card
2. On a Post Page, the link preview card always shows (if there is one). **There is no plus/minus expand/collapse icon on the post page.*

## After

### Listing Page Unexpanded

<img width="798" alt="Screenshot 2023-06-26 at 8 14 41 PM" src="https://github.com/LemmyNet/lemmy-ui/assets/643417/cbd98116-751b-4a09-b630-05af20963e25">


### Listing Page Expanded

<img width="822" alt="Screenshot 2023-06-26 at 8 14 54 PM" src="https://github.com/LemmyNet/lemmy-ui/assets/643417/c539951e-8b73-42e7-9b81-2187437689c8">



### Post Page Unexpanded

<img width="820" alt="Screenshot 2023-06-26 at 8 14 30 PM" src="https://github.com/LemmyNet/lemmy-ui/assets/643417/67660ddf-cbcc-4731-9ced-0851985d9584">

SleeplessOne1917 reviewed (approved):
I prefer it without the body preview.

jsit commented:
Waiting to hear from @alectrocute @dessalines

alectrocute commented:
@jsit Which one’s the post body toggle? The book icon? It’s not immediately clear to me how to preview the body. Maybe that’s something that can be addressed in this PR?

Either way, I don’t like the post body previews either.

jsit commented:
@alectrocute Yeah, I'd like to get it right in this PR. With images, there's a plus button. It seems like there are three things that need a button:

1. Expand image
2. Expand body
3. Show link preview text

Which button should we use for each of these, and where should it be? Should they all be the same?

jsit commented:
@alectrocute It might be best to get rid of that "book" icon (was always confusing to me) and put everything behind a "plus/minus" icon.

alectrocute commented:
> @alectrocute Yeah, I'd like to get it right in this PR. With images, there's a plus button. It seems like there are three things that need a button:
> 
> 1. Expand image
> 2. Expand body
> 3. Show link preview text
> 
> Which button should we use for each of these, and where should it be? Should they all be the same?

Yep, right now it's all in `PostListing`'s `thumbnail()` return. It'd be nice to refactor that, make it a lot simpler and have it be responsible for text-only post body previews (or just expand the whole entire body, why not?) as well as image, video, embed (eg. YouTube) previews.

The book icon should probably be sunset. It doesn't even have a tooltip/label.

alectrocute commented:
It would give me great joy to see `PostListing` be refactored and greatly simplified.

jsit commented:
@alectrocute I've been focusing on it a lot, it's definitely needs to be broken up.

jsit commented:
@alectrocute @SleeplessOne1917 Changed the behavior here and updated the description to show that.

SleeplessOne1917 commented on src/shared/components/post/post-listing.tsx:
```diff
@@ -1780,6 +1755,10 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     setupTippy();
   }
 
+  // handleShowPreview(i: PostListing) {
```
Any reason to keep this commented out function?

jsit replied:
Ah, no -- was planning ahead. Deleted.

jsit commented:
@alectrocute @SleeplessOne1917 One final though -- on a Post page, do we *want* the "preview card" to be behind an expand click? Or could we just always show it?

dessalines commented:
> do we want the "preview card" to be behind an expand click? Or could we just always show it?

IMO the metadata-card should always be shown on the post page, and doesn't need a button.. For example, in github release posts (where I usually have a link, but no post body), the metadata-card automatically shows a few lines of the changelog. Metadata cards aren't usually that tall, and should be summaries. IMO we shouldn't really care too much about vertical height on the post page, because post bodies can already be really long.

This was already referenced, but yes these two buttons conflict, and its not especially clear which one does which:

![Screenshot_2023-06-26-15-03-26-072_mark via gp](https://github.com/LemmyNet/lemmy-ui/assets/930215/7fc2897d-d2a5-4c3d-9655-1d2949cd5006)

jsit commented:
@dessalines Great, thanks for the feedback. I'll work on this PR a bit to get things to work the way you described.

jsit commented:
@dessalines @alectrocute I've changed the behavior and updated the description. The expand/collapse button will only show on "listing" pages, and will show both the post body and the link preview card.
